.docs/content/1.concepts/7.adaptors.md

```diff
@@ -0,0 +1,71 @@
+# Adaptors
+
+The adaptors projects provide implementations of the interfaces relying on different kind of services.
+We plan to provide an implementation for on-premises installation and each cloud provider.
+They are used as abstrations for:
+
+- The **queue** whose role is to distribute messages representing tasks to execute to polling agents.
+- The **database** which is used to store the metadata of the tasks, tasks input and/or output data, partitions and authentication.
+It keeps track of the state of the computations and data managed by ArmoniK.
+- The **object storage** where the data used as task input and/or output is stored.
+
+## Adaptor dynamic loading
+
+Interfaces that are implemented by the adaptors are located in `ArmoniK.Core.Base` project.
+They will be loaded dynamically during startup (today, only the queue can be loaded dynamically but the other adpters will come soon).
+The `ArmoniK.Core.Base` project also generate a (Nuget package)[https://www.nuget.org/packages/ArmoniK.Core.Base] with the interfaces and classes needed for the implementation of a custom adaptor.
+
+Adaptors must implement the `IDependencyInjectionBuildable` as the class that implement this interface will be instantiated and used to build the instances that implement the adaptor interfaces.
+This interface exposes a `Build` method that will be integrated in the dependency injection.
+It will allow adaptors to instantiate their classes and expose the implemented interfaces through dependency injection.
+Then, the interfaces will be used by the middleware to interact with the different storages.
+
+## Class initialization and health check
+
+Adaptor interfaces inherit from the `IInitializable` that requires an `Init` and a `Check` methods.
+Therefore, it is mandatory to implement them.
+`Init` is used for deferred initialization of attributes in class instances (for instance, the creation of a connection after the build of the instances through the Dependency Injection).
+`Check` provides an health check interface used to request the status of the adpater (for instance, are we still connected to the queue ? Can we still receive messages ?).
+These adaptor health checks form the health check of the application (polling agents, control plane, dependency checker, metrics exporters).
+
+## Queue Adaptors
+
+Queue adaptors have to implement `IPullQueueStorage` and `IPushQueueStorage` interfaces for message retrieving from and insertion into the queue.
+
+Pre-included queue adaptors are also available such as:
+
+- `ArmoniK.Core.Adapters.Amqp` provides an AMQP connector for ArmoniK.
+It allows using any AMQP server such as ActiveMQ, ActiveMQ Artemis, RabbitMQ or IBM MQ
```

Comment:
```suggestion
It allows using any AMQP server such as ActiveMQ Artemis, RabbitMQ or IBM MQ
```

Reply:
There is an ActiveMQ and an ActiveMQ Artemis (this is the new version)